File: benchmarks/NSubstitute.Analyzers.Benchmarks.Source.CSharp/DiagnosticsSources/AsyncReceivedInOrderCallbackDiagnosticsSource.cs
using System.Threading.Tasks;

namespace NSubstitute.Analyzers.Benchmarks.Source.CSharp.DiagnosticsSources
{
    public class AsyncReceivedInOrderCallbackDiagnosticsSource
    {
        public void NS5002_AsyncCallbackUsedInReceivedInOrderMethod()
        {
            Received.InOrder(async () => { await Task.CompletedTask; });
        }
    }
}

File: benchmarks/NSubstitute.Analyzers.Benchmarks.Source.VisualBasic/DiagnosticsSources/AsyncReceivedInOrderCallbackDiagnosticsSource.vb
Imports System.Threading.Tasks

Namespace DiagnosticsSources
    Public Class AsyncReceivedInOrderCallbackDiagnosticsSource
        Public Sub NS5002_AsyncCallbackUsedInReceivedInOrderMethod()
            NSubstitute.Received.InOrder(Async Sub()
                Await Task.CompletedTask
            End Sub)
        End Sub
    End Class
End Namespace
